src/halfutils.c

```diff
@@ -225,6 +327,23 @@ SupportsCpuFeature(unsigned int feature)
 	/* Now check features */
 	return (exx[2] & feature) == feature;
 }
+
+#ifdef HAVE_AVX512FP16
+TARGET_XSAVE static bool
+SupportsAvx512Fp16()
+{
+	unsigned int exx[4] = {0, 0, 0, 0};
+	unsigned int feature = (1 << 23);
+
+#if defined(HAVE__GET_CPUID)
+	__get_cpuid_count(7, 0, &exx[0], &exx[1], &exx[2], &exx[3]);
+#elif defined(HAVE__CPUID)
+	__cpuid(exx, 7, 0);
+#endif
+
+	return (exx[3] & feature) == feature;
+}
+#endif
```

Comment:
I think this is missing a couple steps, such as checking for osxsave and verifying the ZMM registers are enabled.  See `SupportsAvx512Popcount()` for an example.

Reply:
Thanks for the reference. I'll add those checks (OSXSAVE and XCR0 control register).